pkg/model/loader.go

```diff
@@ -58,24 +84,23 @@ func (ml *ModelLoader) ListModels() ([]string, error) {
 	return models, nil
 }
 
-func (ml *ModelLoader) TemplatePrefix(modelName string, in interface{}) (string, error) {
-	ml.mu.Lock()
-	defer ml.mu.Unlock()
+// TODO: Split ModelLoader and TemplateLoader? Just to keep things more organized. Left together to share a mutex until I look into that.
 
-	m, ok := ml.promptsTemplates[modelName]
-	if !ok {
-		modelFile := filepath.Join(ml.ModelPath, modelName)
-		if err := ml.loadTemplateIfExists(modelName, modelFile); err != nil {
-			return "", err
-		}
+func evaluateTemplate[T any](templateName string, in T, modelPath string, templateMap *(map[string]*template.Template), mutex *sync.Mutex) (string, error) {
```

Comment:
any specific need for the generics here? I'd keep them only when strictly needed otherwise it is just confusing (what's the catch instead of using an interface or `any` ?)

Reply:
Hey @mudler ! I'm still catching on to `go` style, but I'll admit I'm a _big_ fan of generics in most cases... probably more than I should be!

I suppose we could rewrite this to `interface{}` instead of T - personally, I think the T approach is clearer to read, as we could point at line 119 for example to see that TemplateForChatMessage expects and only works with ChatMesageTemplate, and not interface{}. Is there another way to express this type of constraint in `go` that I've overlooked?

Reply:
I'm fine either ways. I'm just a bit not super convinced in this case because raises code complexity (see e.g. pointer of maps) for no real catch